pinot-spi/src/main/java/org/apache/pinot/spi/config/instance/Instance.java

```diff
@@ -38,29 +38,40 @@
  *   "tags": ["example_OFFLINE"],
  *   "pools": {
  *     "example_OFFLINE": 0
- *   }
+ *   },
+ *   "grpcPort": 8090
  * }
  * </pre>
  */
 public class Instance extends BaseJsonConfig {
+  public static int NOT_SET_GRPC_PORT_VALUE = -1;
+
   private final String _host;
   private final int _port;
   private final InstanceType _type;
   private final List<String> _tags;
   private final Map<String, Integer> _pools;
+  private final int _grpcPort;
+
+
+  public Instance(String host, int port, InstanceType type, List<String> tags, Map<String, Integer> pools) {
+    this(host, port, type, tags, pools, NOT_SET_GRPC_PORT_VALUE);
+  }
 
   @JsonCreator
   public Instance(@JsonProperty(value = "host", required = true) String host,
       @JsonProperty(value = "port", required = true) int port,
       @JsonProperty(value = "type", required = true) InstanceType type,
-      @JsonProperty("tags") @Nullable List<String> tags, @JsonProperty("pools") @Nullable Map<String, Integer> pools) {
+      @JsonProperty("tags") @Nullable List<String> tags, @JsonProperty("pools") @Nullable Map<String, Integer> pools,
+      @JsonProperty("grpcPort") int grpcPort) {
     Preconditions.checkArgument(host != null, "'host' must be configured");
     Preconditions.checkArgument(type != null, "'type' must be configured");
     _host = host;
     _port = port;
     _type = type;
     _tags = tags;
     _pools = pools;
+    _grpcPort = grpcPort;
```

Comment:
This will be set to 0 from the json if `grpcPort` is not set. You can check for 0 and put -1